vowpalwabbit/core/include/vw/core/parse_example_json.h

```diff
@@ -967,14 +967,17 @@ class DefaultState : public BaseState<audit>
           *p = '_';
       }
     }
-
-    if (ctx._chain_hash) { ctx.CurrentNamespace().AddFeature(ctx.key, str, ctx._hash_func, ctx._parse_mask); }
-    else
+    const char* ns = ctx.CurrentNamespace().name;
+    if (ctx.ignore_features == nullptr || !(ctx.ignore_features->find(ns) != ctx.ignore_features->end() && ctx.ignore_features->at(ns).find(ctx.key) != ctx.ignore_features->at(ns).end()))
```

Comment:
nit:

replace all the negativity with some positivity? might be easier to read

```suggestion
    if (ctx.ignore_features == nullptr || (ctx.ignore_features->find(ns) == ctx.ignore_features->end() || (ctx.ignore_features ctx.ignore_features->at(ns).find(ctx.key) == ctx.ignore_features->at(ns).end()))
```

Reply:
good point